clang/lib/CodeGen/CGLoopInfo.h

```diff
@@ -160,6 +160,12 @@ struct LoopAttributes {
   /// Flag for llvm.loop.fusion.disable metatdata.
   bool SYCLNofusionEnable;
 
+  /// Flag for llvm.loop.intel.pipelining.enable metadata.
+  bool SYCLIntelFPGAPipelineEnable;
```

Comment:
Same question here -- why do we need two variables?

Reply:
Instead of having an `unsigned` and `bool` here, I'd have `Optional<bool> SYCLIntelFPGAPipelineEnable = llvm::None;`. When you go to explicitly enable/disable, then it'd set the optional to either `true` or `false`. Does that make sense?

Reply:
Added single var to cover this part.